### PR TITLE
mardizzone/pos-649

### PR DESCRIPTION
**Pull request by marcello33**
# Description

This PR implements two new features:

- A blueprint for the CI: the goal is to use `express-cli` with `remote setup` to deploy devnets from the CI and run smoke tests. This is complementary to `bor` and `heimdall` CIs, which use `matic-cli` and `docker setup`, to have a 100% coverage on testing toolkit functionalities across Polygon stack.
The `integration-tests` are currently mocked as devOps suggested a new way to implement them, using their own GH action (will get back to it once this is merged).

- Replace usage of `aws key pairs` with `aws sso`. This removes the long-lived shared tokens, which carry security burdens. Please, follow the instructions in the `README` and update your own `.env` file.   

# Changes

- [ ] Bugfix (non-breaking change that solves an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [x] Breaking change (change that is not backwards-compatible and/or changes current functionality)
- [ ] New test case for remote devnet

# Checklist

- [x] I have added at least 2 reviewer or the whole pos-v1 team
- [x] I have added sufficient documentation in code
- [x] I will be resolving comments - if any - by pushing each fix in a separate commit and linking the commit hash in the comment reply

## Testing

- [x] I have tested this code manually on local environment
- [x] I have tested this code manually on remote devnet using express-cli
- [ ] I have tested this code manually on mumbai

### Manual tests

Tested on CI too

### Breaking changes

Given this changes the authentication process between `terraform` and `aws`, you need to follow the new instructions in the `README` file, with paying attention to the `Auth Configuration` section.
This also implies that your `.env` file must be updated, otherwise you will get unauthorized error when interacting with remote devnets.

### CODE REVIEW

**Review by Raneet10 — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+          sed -i 's,aws-key,matic-cli-ci-key,' .env
+          sed -i 's,/absolute/path/to/your/,/home/runner/work/matic-cli/matic-cli/matic-cli/aws/,' .env
+          sed -i 's,MATIC_CLI_BRANCH=master,MATIC_CLI_BRANCH=${{ github.head_ref || github.ref_name }},' .env
+          sed -i 's,DATADOG_API_KEY,TODO,' .env
+          cat .env
+
+      - name: Setup aws
+        continue-on-error: true
+        run: |
+          cd matic-cli
```
Comment (JekaMas) [inline .github/workflows/ci.yml:45]: one day we'd need to make the same with ansible

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+          mkdir -p aws
+          curl "https://awscli.amazonaws.com/awscli-exe-linux-x86_64.zip" -o "awscliv2.zip"
+          unzip awscliv2.zip
+          sudo ./aws/install --update
+          aws configure set aws_access_key_id TODO
```
Comment (JekaMas) [inline .github/workflows/ci.yml:50]: are these TODOs required?

**Review by JekaMas — APPROVED** [review]

File: .github/workflows/ci.yml (hunk 1 above)
Comment (marcello33) [reply .github/workflows/ci.yml:45]: Yes :)

File: .github/workflows/ci.yml (hunk 2 above)
Comment (marcello33) [reply .github/workflows/ci.yml:50]: Well, I will need to replace them with GH secrets but have no access to it. Will ask Sandeep.
Also, this whole CI will be integrated with GH action to secure terraform-aws authentication, as shown here https://www.notion.so/polygontechnology/No-More-Secrets-6382754faac745aa9bd049a3ed423218

**Pull request merged**